parser/types.go

```diff
@@ -119,10 +119,18 @@ func incompletePipeExpr(call interface{}, text []byte, pos position) (*ast.PipeE
 func memberexprs(head, tail interface{}, text []byte, pos position) (ast.Expression, error) {
 	res := head.(ast.Expression)
 	for _, prop := range toIfaceSlice(tail) {
-		res = &ast.MemberExpression{
-			Object:   res,
-			Property: prop.(ast.Expression),
-			BaseNode: base(text, pos),
+		if integer, ok := prop.(*ast.IntegerLiteral); ok {
```

Comment:
What happens when you have an expression that returns an integer?

Reply:
Good catch. I did not consider this case. I think the proper way to handle it is probably to have a single `MemberExpression` AST node and have type inference figure out if it represents array access or object access. This will mean re-working a lot of the changes here.

Reply:
👍 

This is likely going to require a lot of reworking. Can we close this PR since this method is unlikely to work and we'll create a new branch/new PR  for the new method? I think what you described will work much better personally.

Reply:
Yes closing this PR.